app/src/main/java/org/oppia/android/app/testing/TextInputInteractionViewTestActivity.kt

```diff
@@ -0,0 +1,112 @@
+import org.oppia.android.app.player.state.listener.StateKeyboardButtonListener
+import org.oppia.android.databinding.ActivityTextInputInteractionViewTestBinding
+import org.oppia.android.util.extensions.getProtoExtra
+import org.oppia.android.util.extensions.putProtoExtra
+import javax.inject.Inject
+
+/**
+ * This is a dummy activity to test input interaction views.
+ * It contains [TextInputInteractionView]
+ */
+class TextInputInteractionViewTestActivity :
+  InjectableAutoLocalizedAppCompatActivity(),
+  StateKeyboardButtonListener,
+  InteractionAnswerErrorOrAvailabilityCheckReceiver,
+  InteractionAnswerReceiver {
+
+  private lateinit var binding: ActivityTextInputInteractionViewTestBinding
+
+  @Inject
+  lateinit var textinputViewModelFactory: TextInputViewModel.FactoryImpl
+
+  /** Gives access to the [TextInputViewModel]. */
+  val textInputViewModel by lazy {
+    textinputViewModelFactory.create<TextInputViewModel>()
+  }
+
+  /** Gives access to the translation context. */
+  lateinit var writtenTranslationContext: WrittenTranslationContext
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    (activityComponent as ActivityComponentImpl).inject(this)
+    binding = DataBindingUtil.setContentView(
+      this, R.layout.activity_text_input_interaction_view_test
+    )
+
+    val params = intent.getProtoExtra(
+      TEST_ACTIVITY_PARAMS_ARGUMENT_KEY, InputInteractionViewTestActivityParams.getDefaultInstance()
+    )
+    writtenTranslationContext = params.writtenTranslationContext
+    binding.textInputViewModel = textInputViewModel
+  }
+
+  /** Checks submit-time errors. */
+  fun getPendingAnswerErrorOnSubmitClick(v: View) {
+    textInputViewModel.checkPendingAnswerError(AnswerErrorCategory.SUBMIT_TIME)
+  }
+
+  override fun onPendingAnswerErrorOrAvailabilityCheck(
+    pendingAnswerError: String?,
+    inputAnswerAvailable: Boolean
+  ) {
+  }
+
+  override fun onAnswerReadyForSubmission(answer: UserAnswer) {
+  }
+
+  override fun onEditorAction(actionCode: Int) {
+  }
+
+  private inline fun <reified T : StateItemViewModel>
+  StateItemViewModel.InteractionItemFactory.create(
+    interaction: Interaction = Interaction.getDefaultInstance()
+  ): T {
+    return create(
+      entityId = "fake_entity_id",
+      hasConversationView = false,
+      interaction = interaction,
+      interactionAnswerReceiver = this@TextInputInteractionViewTestActivity,
+      answerErrorReceiver = this@TextInputInteractionViewTestActivity,
+      hasPreviousButton = false,
+      isSplitView = false,
+      writtenTranslationContext,
+      timeToStartNoticeAnimationMs = null
+    ) as T
+  }
+
+  companion object {
+    private const val TEST_ACTIVITY_PARAMS_ARGUMENT_KEY =
+      "TextInputInteractionViewTestActivity.params"
+
+    /** Creates an intent to open this activity. */
+    fun createIntent(
+      context: Context,
+      extras: InputInteractionViewTestActivityParams
```

Comment:
Create params for `TextInputInteractionViewTestActivity`

Reply:
I don't see any use of creating intent, so removed.